protocol/messenger_share_urls.go

```diff
@@ -511,13 +523,17 @@ func (m *Messenger) parseUserURLWithData(data string, chatKey string) (*URLDataR
 	}, nil
 }
 
-func (m *Messenger) IsStatusSharedURL(url string) bool {
-	return strings.HasPrefix(url, baseShareURL)
+func IsStatusSharedURL(url string) bool {
+	return strings.HasPrefix(url, sharedURLUserPrefix) ||
+		strings.HasPrefix(url, sharedURLUserPrefixWithData) ||
+		strings.HasPrefix(url, sharedURLCommunityPrefix) ||
+		strings.HasPrefix(url, sharedURLCommunityPrefixWithData) ||
+		strings.HasPrefix(url, sharedURLChannelPrefixWithData)
 }
 
 func (m *Messenger) ParseSharedURL(url string) (*URLDataResponse, error) {
-	if !m.IsStatusSharedURL(url) {
-		return nil, fmt.Errorf("url should start with '%s'", baseShareURL)
+	if !IsStatusSharedURL(url) {
```

Comment:
it would be a little more obvious to my taste if you combine here checking and parsing urls, something like this:
```
	if chatKey := chatKeyFromUrl(url); isStatusCommunityUrl(url) {
		return m.parseCommunityURLWithChatKey(chatKey)
	}
    ...
```

Reply:
nice idea!
refactored the function, looks better to me now 👍